pkg/controller/periodicity/periodicity_controller.go

```diff
@@ -94,29 +86,35 @@ func (c *Controller) syncStatefulSetTimeStamp() error {
 	if err != nil {
 		return err
 	}
+
 	var errs []error
 	for _, sts := range stsList {
 		// If there is any error during our sts annotation updating, we just collect the error
 		// and continue to next sts
-		ok, tcRef := util.IsOwnedByTidbCluster(sts)
+		ok, _ := util.IsOwnedByTidbCluster(sts)
 		if !ok {
 			continue
 		}
-		tc, err := c.tcLister.TidbClusters(sts.Namespace).Get(tcRef.Name)
-		if err != nil {
-			errs = append(errs, err)
-			continue
-		}
 		if sts.Annotations == nil {
 			sts.Annotations = map[string]string{}
 		}
 		sts.Annotations[label.AnnStsLastSyncTimestamp] = time.Now().Format(time.RFC3339)
-		newSts, err := c.statefulSetControl.UpdateStatefulSet(tc, sts)
+		var mergePatch []byte
+		mergePatch, err = json.Marshal(map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"annotations": sts.Annotations,
+			},
+		})
```

Comment:
I think using the Patch method is a good idea. It can avoid API conflicts in kube-apiserver.

But the root cause is we should copy before modifying the statefulset.